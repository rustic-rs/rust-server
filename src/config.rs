pub mod auth_config;
pub mod auth_file;
pub mod server_config;
