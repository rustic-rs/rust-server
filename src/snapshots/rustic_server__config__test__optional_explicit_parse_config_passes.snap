---
source: src/config.rs
expression: config
---
RusticServerConfig {
    server: ConnectionSettings {
        listen: Some(
            127.0.0.1:8000,
        ),
    },
    storage: StorageSettings {
        data_dir: Some(
            "./test_data/test_repos/",
        ),
        quota: None,
    },
    auth: HtpasswdSettings {
        disable_auth: true,
        htpasswd_file: None,
    },
    acl: AclSettings {
        disable_acl: true,
        private_repos: true,
        append_only: true,
        acl_path: None,
    },
    tls: TlsSettings {
        disable_tls: true,
        tls_key: None,
        tls_cert: None,
    },
    log: LogSettings {
        log_level: Some(
            "info",
        ),
        log_file: None,
    },
}
